### PR TITLE
Adding support for more generic dictionary entry slots for R2R

### PR DESCRIPTION
**Pull request by fadimounir**
Adding support to the following generic dictionary entry slots for R2R:
	DeclaringTypeHandleSlot
	ConstrainedMethodEntrySlot

### CODE REVIEW

**Comment by fadimounir** [comment]
@dotnet-bot test Ubuntu Checked pri1r2r please

**Comment by fadimounir** [comment]
@dotnet-bot test Windows_NT Checked pri1r2r please

**Comment by fadimounir** [comment]
@dotnet-bot test Ubuntu Checked pri1r2r please

**Comment by fadimounir** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build please

**Comment by fadimounir** [comment]
@dotnet-bot test Linux ARM Emulator Cross Release Build please

**Comment by fadimounir** [comment]
@JohnChen0 @jkotas PTAL

File: src/inc/readytorun.h [hunk 1]
```diff
@@ -166,6 +166,7 @@ enum ReadyToRunFixupKind
     READYTORUN_FIXUP_Check_FieldOffset          = 0x2B,
 
     READYTORUN_FIXUP_DelegateCtor               = 0x2C, /* optimized delegate ctor */
+    READYTORUN_FIXUP_DeclaringTypeHandle        = 0x2D,
 };
```
Comment (jkotas) [inline src/inc/readytorun.h]: Do we need to bump the version number as was discussed?

**Comment by jkotas** [comment]
Any interesting test cases that should be added to the R2R unit test to exercise this?

**Comment by jkotas** [comment]
LGTM otherwise.

**Comment by fadimounir** [comment]
@jkotas I am planning to merge these changes after merging the FieldDescHandle changes, which include the version change (see other PR). 
For tests, I am considering adding a small collection of TypeGenerator tests, which cover interesting areas with generics, including the changes in this PR. I will do that once I get back from vacation

**Comment by fadimounir** [comment]
@dotnet-bot test Windows_NT Checked pri1r2r please

**Comment by fadimounir** [comment]
@dotnet-bot test Ubuntu Checked pri1r2r please

**Comment by fadimounir** [comment]
@dotnet-bot test Ubuntu Checked pri1r2r please

**Pull request merged**